### PR TITLE
MultiFromOutputStream blocking close refactor

### PR DESCRIPTION
**Pull request by danielkec**
Fixes https://gitlab-odx.oracledx.com/helidon/helidon/-/issues/18

Signed-off-by: Daniel Kec <daniel.kec@oracle.com>

### CODE REVIEW

File: common/reactive/src/main/java/io/helidon/common/reactive/MultiFromOutputStream.java [hunk 1]
```diff
@@ -116,48 +102,6 @@ public void write(int b) throws IOException {
...
-     * publishing output stream {@code close()} method.
-     * <p>
-     * Additionally, this mechanism can be used to propagate any downstream completion exceptions back to this publisher
-     * and up it's call stack. When a non-null {@code throwable} parameter is passed into the method, it will be wrapped
-     * in an {@link IOException} and thrown from the {@code close()} method when it is invoked.
-     *
-     * @param throwable represents a completion error condition that should be thrown when a {@code close()} method is invoked
-     *                  on this publishing output stream. If set to {@code null}, the {@code close()} method will exit normally.
-     */
-    void signalCloseComplete(Throwable throwable) {
```
Comment (danielkec) [inline common/reactive/src/main/java/io/helidon/common/reactive/MultiFromOutputStream.java:155]: @romain-grecourt Is this still actual? There is no exception downstream can propagate to upstream except NPO, also is blocking of close still neccessary? Only proper way to complete is from upstream's side or maybe wait for cancel from downstream, but in such case we cant complete from upstream with close which is blocked by waiting for cancel.

Comment (romain-grecourt) [reply common/reactive/src/main/java/io/helidon/common/reactive/MultiFromOutputStream.java:155]: Not sure why you are asking me, did I write this code ? :D

Comment (danielkec) [reply common/reactive/src/main/java/io/helidon/common/reactive/MultiFromOutputStream.java:155]: Sorry I was following the blame :detective: . Next one to ask is @tomas-langer who modified the file last, WDYT Tomas?

**Comment by danielkec** [comment]
@tomas-langer I have cherrypicked https://gitlab-odx.oracledx.com/helidon/helidon/-/issues/18 fix to #1991 so it can be targeted for 2.0, this PR can wait post 2.0 then

**Pull request merged**